File: frontend/src/types/structure.ts
export interface AtomData {
  index: number;
  symbol: string;
  x: number;
  y: number;
  z: number;
  color: string;
  radius: number;
}

export interface BondData {
  atom1: number;
  atom2: number;
  order: number;
}

export interface MoleculeData {
  formula: string;
  name: string;
  atoms: AtomData[];
  bonds: BondData[];
  properties: Record<string, unknown>;
}

export interface OrbitalInfo {
  n: number;
  l: number;
  label: string;
  electrons: number;
  shape: string;
  radius: number;
  orientations: string[];
}

export interface OrbitalData {
  element: string;
  atomic_number: number;
  electron_configuration: string;
  orbitals: OrbitalInfo[];
}

File: backend/app/models/structure.py
from pydantic import BaseModel


class AtomData(BaseModel):
    index: int
    symbol: str
    x: float
    y: float
    z: float
    color: str
    radius: float


class BondData(BaseModel):
    atom1: int
    atom2: int
    order: int


class MoleculeData(BaseModel):
    formula: str
    name: str
    atoms: list[AtomData]
    bonds: list[BondData]
    properties: dict = {}


class OrbitalInfo(BaseModel):
    n: int
    l: int
    label: str
    electrons: int
    shape: str
    radius: float
    orientations: list[str] = []


class OrbitalData(BaseModel):
    element: str
    atomic_number: int
    electron_configuration: str
    orbitals: list[OrbitalInfo]
